### PR TITLE
multi: Use setaltsig with trezor ticket purchase.

### PR DESCRIPTION
**Pull request by JoeGruffins**
multi: Use setaltsig with trezor ticket purchase.
    
    Use several new dcrwallet grpc methods in order to use the seed from
    trezor in order to create a voting xpriv. Further use them to retrieve
    the voting key and signing address from the wallet for ticket.

    grpc: Update wallet methods.
    
    Add Address, DumpPrivateKey, and ImportVotingAccountFromSeed. Add voting
    account parameters to PurchaseTicketsRequest.

### CODE REVIEW

**Comment by JoeGruffins** [comment]
This is still far from mainnet, but adds more logic needed to purchase tickets and sign messages to the vspd using another deterministic key.

This requires a vsp running https://github.com/decred/vspd/pull/287 I have it running on dcrstakedinner.com and if there any problems using that vspd, please @ me.
It also requires dcrwallet running https://github.com/decred/decrediton/pull/3533 which implements all of the new grpc calls added.

Very open to any suggestions.

File: app/actions/TrezorActions.js [hunk 1]
```diff
@@ -1152,12 +1194,24 @@ export const purchaseTicketsV3 = (accountNum, numTickets, vsp) => async (
...
         chainParams
       );
+      const votingAddr = decodedTicket.outputs[0].decodedScript.address;
+      // Get address's index.
+      const addrValidResp = await wallet.validateAddress(
+        walletService,
+        votingAddr
+      );
+      const addrIdx = addrValidResp.index;
+      // The signing address is the external account with the same index.
```
Comment (JoeGruffins) [inline app/actions/TrezorActions.js:1209]: external branch

File: app/actions/TrezorActions.js [hunk 2]
```diff
@@ -1202,6 +1269,45 @@ export const purchaseTicketsV3 = (accountNum, numTickets, vsp) => async (
...
+    tickethex: txHex,
+    parenthex: parentTxHex,
+    altpubaddress: signingAddr
+  };
+  const jsonStr = JSON.stringify(req);
+  const sig = await signMessageAttemptTrezor(commitmentAddr, jsonStr)(
+    dispatch,
+    getState
+  );
+  if (!sig) throw "unable to sign fee address message";
```
Comment (JoeGruffins) [inline app/actions/TrezorActions.js:1306]: set alternate signature message

**Comment by matheusd** [comment]
Looks overall reasonable so far (modulo all the outstanding stuff :P)

**Comment by JoeGruffins** [comment]
Currently not working with dcrwallet on master because of different grpc versions, but can be used with https://github.com/JoeGruffins/dcrwallet/tree/useaccoutforvoting.flat

**Comment by alexlyp** [comment]
Any status on this? It's been in draft since June 2021.

**Comment by JoeGruffins** [comment]
Same comment as https://github.com/decred/dcrwallet/pull/1946

tldr: waiting for dcrwallet

**Comment by alexlyp** [comment]
Same as from my comment on the Trezor purchase ticket PR.
